### PR TITLE
Floor hazards! Mouse traps & Glass Shards.

### LINKED ISSUE #5754: Floor Hazards - Watch your step!
### Description

- [x] Glass shards, when walked on should deal damage to peoples' feet if they are not protected by shoes. 
- [x] Mouse traps should be armable and placeable to deal with cable-chewing mice or unobservant crewmembers
- [x] armed mouse traps when placed inside a container should snap at anyone who opens the bag _one at a time,_ unless the person opens the bag on disarm intent.
- [x] igniters, grenades, and remote signallers should be attacheable to mouse traps, and should trigger when the mouse trap is triggered.

### PR DESCRIPTION
**Pull request by craftxbox**
### Purpose
makes progress on #5754 
Glass shards will now shred feet, mousetraps will snap on mice, and also feet, sometimes your hands if you arent careful.
TODO:
Mousetraps to trigger igniters, grenades and remote signallers. and be able to be layed as traps inside bags/etc

### Notes:
Draft PR. Mousetraps and shards will currently destroy your feet if you're not looking. Not sure exactly how to hook into inventory opens for the latter parts of the issue.
My main thought was to put all the code to deal with mousetraps going off in inventory in the `TriggerInventoryTrap` method here, but it's just empty right now because... yeah. I feel very bad that it's taken me this long to get to this point.

### Changelog:
CL: Glass shards now destroy your feet if you don't wear shoes.
CL: Mousetraps are now capable of snapping on mice. And your feet.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/FloorHazard.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public void ArmHazard(){
+        armed = true;
+    }
+
+    public void DisarmHazard(){
+        armed = false;
+    }
+
+    public void OnEnterableEnter(BaseEventData eventData)
+	{
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/FloorHazard.cs:30]: Could you start the { on the next line for all the scripts, just looks better and less cluttered imo. Also adding empty lines to split up code sections doesnt cost performance :p

Also the OnEnterableEnter seems to have weird indenting

File: UnityProject/Assets/Scripts/Objects/MouseTrap.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            } else if (eventData.selectedObject.TryGetComponent(out PlayerScript _)){
+                floorHazard.OnEnterableEnter(eventData);
+            } else return;
+            armed = false;
+        }
+    }
+
+    public void TriggerInventoryTrap(BaseEventData eventData){
+
+    }
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/MouseTrap.cs:58]: Is this supposed to be empty?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/MouseTrap.cs:58]: Oh wait nvm thats the TODO

File: UnityProject/Assets/Scripts/Items/FloorHazard.cs (hunk 1 above)
Comment (craftxbox) [reply UnityProject/Assets/Scripts/Items/FloorHazard.cs:30]: yeahh, i wont lie i dont really think about indentation alot

Comment (jblevins1991) [reply UnityProject/Assets/Scripts/Items/FloorHazard.cs:30]: Can we get a linter and have some github hooks auto format on commit? These nit pick comments on PRs don't provide value to the review.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/FloorHazard.cs:30]: set something up and pr it

Comment (craftxbox) [reply UnityProject/Assets/Scripts/Items/FloorHazard.cs:30]: > Can we get a linter and have some github hooks auto format on commit? These nit pick comments on PRs don't provide value to the review.

they're not valuable code-wise but i still find them valuable

File: UnityProject/Assets/Scripts/Items/FloorHazard.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if(!requiresArming || armed){
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/FloorHazard.cs:31]: extract and encapsulate this logic into different methods. Too much nesting and complexity on this method alone.

**Comment by NoooneyDude** [comment]
If you want to fix the merge conflicts (cleaning up the style would be nice too), then I think this is useful to merge (after freeze is lifted) and you can complete the inventory functionality at your own leisure.

Else if you've abandoned the PR or have no time for it, just let us know and I'm sure someone else will fix it up for merging.

**Comment by vlandemart** [comment]
Closing Abandoned PR.
In case anyone resumes work on this - just reopen it.

**Pull request closed without merging**